ultralytics/yolo/utils/callbacks/clearml.py

```diff
@@ -12,49 +17,104 @@
+        """
+    for f in files:
+        if f.exists():
+            it = re.search(r'_batch(\d+)', f.name)
+            iteration = int(it.groups()[0]) if it else 0
+            Task.current_task().get_logger().report_image(title=title,
+                                                          series=f.name.replace(it.group(), ''),
+                                                          local_path=str(f),
+                                                          iteration=iteration)
+
+
+def _log_plot(title, plot_path):
+    """
+        Log image as plot in the plot section of ClearML
+
+        arguments:
+        title (str) Title of the plot
+        plot_path (PosixPath or str) Path to the saved image file
+        """
+    img = mpimg.imread(plot_path)
+    fig = plt.figure()
+    ax = fig.add_axes([0, 0, 1, 1], frameon=False, aspect='auto', xticks=[], yticks=[])  # no ticks
+    ax.imshow(img)
+
+    Task.current_task().get_logger().report_matplotlib_figure(title, '', figure=fig, report_interactive=False)
 
 
 def on_pretrain_routine_start(trainer):
+    # TODO: reuse existing task
     try:
-        task = Task.init(project_name=trainer.args.project or 'YOLOv8',
-                         task_name=trainer.args.name,
-                         tags=['YOLOv8'],
-                         output_uri=True,
-                         reuse_last_task_id=False,
-                         auto_connect_frameworks={'pytorch': False})
+        if Task.current_task():
+            task = Task.current_task()
+        else:
+            task = Task.init(project_name=trainer.args.project or 'YOLOv8',
+                             task_name=trainer.args.name,
+                             tags=['YOLOv8'],
+                             output_uri=True,
+                             reuse_last_task_id=False,
+                             auto_connect_frameworks={
+                                 'pytorch': False,
+                                 'matplotlib': False})
+            LOGGER.warning(
+                f'ClearML Initialized a new task. If you want to run remotely, please add clearml-init and connect your arguments before initializing YOLO.'
+            )
         task.connect(vars(trainer.args), name='General')
     except Exception as e:
         LOGGER.warning(f'WARNING ⚠️ ClearML installed but not initialized correctly, not logging this run. {e}')
 
 
 def on_train_epoch_end(trainer):
     if trainer.epoch == 1:
-        _log_images({f.stem: str(f) for f in trainer.save_dir.glob('train_batch*.jpg')}, 'Mosaic', trainer.epoch)
+        _log_debug_samples(sorted(trainer.save_dir.glob('train_batch*.jpg')), 'Mosaic')
 
 
 def on_fit_epoch_end(trainer):
-    task = Task.current_task()
-    if task and trainer.epoch == 0:
+    # You should have access to the validation bboxes under jdict
+    Task.current_task().get_logger().report_scalar('Epoch Time',
+                                                   'Epoch Time',
```

Comment:
@thepycoder is this duplicated input correct?

Reply:
Yes it is, it is meant as a title and a series name. In this case there is only 1 series on the graph so I tend to call them the same. Series can also be an empty string, but it is a required parameter.

I'm updating the code with keyword arguments, so it's at least a little more readable :)